### DISCUSSION TITLE
Does pyo3 support PEP 489 Multi-phase extension module initialization?

### DISCUSSION THREAD

**Discussion by messense** [PyO3/pyo3#1982] [Questions] [answered]
Saw https://twitter.com/VictorStinner/status/1458060898321551370 recently, I wonder does pyo3 support [PEP 489](https://www.python.org/dev/peps/pep-0489/)? Quick search for `PyModuleDef_Init` shows no usage in pyo3's code base.

**Comment by davidhewitt** [comment, marked as answer]
I don't believe we do, however I would really like us to explore this because I think it'll begin to open possibility of mutliple sub-interpreters in PyO3?
